pkg/antctl/raw/set/command.go

```diff
@@ -0,0 +1,181 @@
+  Enable flow-collector
+  $ antctl set flowcollector.enable=true
+`, "\n")
+
+func init() {
+	Command = &cobra.Command{
+		Use:   "set",
+		Short: "Update clickhouse/flow-collector parameters used in flow aggregator",
+	}
+	mutators = map[string]FlowAggregatorConfigMutator{
+		"clickhouse.enable": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setBoolOrFail(&c.ClickHouse.Enable, value)
+		},
+		"clickhouse.database": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setStringOrFail(&c.ClickHouse.Database, value)
+		},
+		"clickhouse.databaseURL": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setStringOrFail(&c.ClickHouse.DatabaseURL, value)
+		},
+		"clickhouse.debug": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setBoolOrFail(&c.ClickHouse.Debug, value)
+		},
+		"clickhouse.compress": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			if c.ClickHouse.Compress == nil {
+				c.ClickHouse.Compress = new(bool)
+			}
+			return setBoolOrFail(c.ClickHouse.Compress, value)
+		},
+		"clickhouse.commitInterval": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setTimeDurationOrFail(&c.ClickHouse.CommitInterval, value)
+		},
+		"clickhouse.ttl": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setIntOrFail(&c.ClickHouse.TTL, value)
+		},
+		"flowcollector.enable": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setBoolOrFail(&c.FlowCollector.Enable, value)
+		},
+		"flowcollector.externalflowcollectoraddr": func(c *flowaggregatorconfig.FlowAggregatorConfig, value string) error {
+			return setStringOrFail(&c.FlowCollector.Address, value)
+		},
+	}
+	Command.Example = example
+	Command.RunE = updateRunE
+}
+
+func updateRunE(cmd *cobra.Command, args []string) error {
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	k8sClient, _, err := raw.SetupClients(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to create clientset: %w", err)
+	}
+	configMap, err := flowaggregatorconfig.GetFAConfigMap(k8sClient)
+	if err != nil {
+		return err
+	}
+	var flowAggregatorConf flowaggregatorconfig.FlowAggregatorConfig
+	if err := yaml.Unmarshal([]byte(configMap.Data["flow-aggregator.conf"]), &flowAggregatorConf); err != nil {
+		klog.Errorf("Failed to unmarshal FlowAggregator config from ConfigMap: %v", err)
+		return err
+	}
+	for _, query := range args {
+		pair := strings.Split(query, "=")
+		if len(pair) != 2 {
+			return fmt.Errorf("query should contains exactly one '='")
+		}
+		if mutatorFn, ok := mutators[pair[0]]; ok {
+			err := mutatorFn(&flowAggregatorConf, pair[1])
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("wrong arg, please check antctl set -h")
```

Comment:
maybe use a more helpful error message here?
Something like "unknown configuration parameter" instead of "wrong arg"?

Reply:
Fixed.